src/NUnitFramework/framework/Compatibility/ReflectionExtensions.cs

```diff
@@ -99,415 +97,127 @@ public static Delegate CreateDelegate(this MethodInfo method, Type type)
-        }
-
-        static bool NotPrivate(MemberInfo info)
+        public static MemberInfo[] GetMembers(this Type type, BindingFlags bindingAttr)
         {
-            var pinfo = info as PropertyInfo;
-            if (pinfo != null)
-                return pinfo.GetMethod?.IsPrivate == false;
-
-            var finfo = info as FieldInfo;
-            if (finfo != null)
-                return finfo.IsPrivate == false;
-
-            var minfo = info as MethodBase;
-            if (minfo != null)
-                return minfo.IsPrivate == false;
-
-            var einfo = info as EventInfo;
-            if (einfo != null)
-                return einfo.RaiseMethod.IsPrivate == false;
-
-            return true;
+            return type.GetTypeInfo().GetMembers(bindingAttr);
         }
 
         /// <summary>
-        /// Gets field of the given name on the type
+        /// Searches for the public field with the specified name.
         /// </summary>
-        /// <param name="type"></param>
-        /// <param name="name"></param>
-        /// <returns></returns>
         public static FieldInfo GetField(this Type type, string name)
         {
-            return type.GetTypeInfo()
-                .DeclaredFields
-                .Where(p => p.Name == name && p.IsPublic)
-                .FirstOrDefault();
+            return type.GetTypeInfo().GetField(name);
         }
 
         /// <summary>
-        /// Gets property of the given name on the type
+        /// Searches for the public property with the specified name.
         /// </summary>
-        /// <param name="type"></param>
-        /// <param name="name"></param>
-        /// <returns></returns>
         public static PropertyInfo GetProperty(this Type type, string name)
         {
-            return type.GetRuntimeProperties()
-                .Where(p => ((p.GetMethod != null && p.GetMethod.IsPublic) || (p.SetMethod != null && p.SetMethod.IsPublic)) && p.Name == name)
-                .FirstOrDefault();
+            return type.GetTypeInfo().GetProperty(name);
         }
 
         /// <summary>
-        /// Gets property of the given name on the type
+        /// Searches for the specified property, using the specified binding constraints.
         /// </summary>
-        /// <param name="type"></param>
-        /// <param name="name"></param>
-        /// <param name="flags"></param>
-        /// <returns></returns>
-        public static PropertyInfo GetProperty(this Type type, string name, BindingFlags flags)
+        public static PropertyInfo GetProperty(this Type type, string name, BindingFlags bindingAttr)
         {
-            bool declaredOnly = flags.HasFlag(BindingFlags.DeclaredOnly);
-            return type.GetRuntimeProperties()
-                .Where(prop => declaredOnly ? prop.DeclaringType.Equals(type) : true)
-                .ApplyBindingFlags(flags)
-                .Where(p => p.Name == name)
-                .FirstOrDefault();
```

Comment:
All of my beautiful Linq code gone! I worked so hard on this 😛

Reply:
😢 On one hand, code is a liability; on the other hand, I know how you feel!

Reply:
I worked for ages getting all of these reflection API's running the same as the built in ones, but I will be happy to drop the complexity 😄

Reply:
I have to say that this is a path I followed for many years myself... trying to hide complexity in order to make life simpler for the average developer. I eventually came to believe that sometimes the complexity needs to be visible, so that people will see it's there and take it into account. But deciding on a case by case basis is rather hard!